include/knowhere/version.h

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2019-2023 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#pragma once
+
+#include <strings.h>
+
+#include <regex>
+
+#include "log.h"
+
+namespace knowhere {
+namespace {
+static constexpr const char* version_regex = R"(^(knowhere)-[vV]\d+\.\d+\.\d+$)";
+static constexpr const char* default_version = "knowhere-v2.2.0";
+static constexpr const char* minimal_vesion = "knowhere-v0.1.0";
+static constexpr const char* current_version = "knowhere-v2.2.0";
+static constexpr size_t version_group_size = 3;
+static constexpr char prefix_delimiter = '-';
+static constexpr char code_delimiter = '.';
+}  // namespace
+
+class Version {
+ public:
+    explicit Version(const char* version_code_) {
+        version_code = version_code_;
+        auto codes = version_split(version_code_);
+        if (codes.size() == version_group_size) {
+            major_version = codes[0];
+            middle_version = codes[1];
+            mini_version = codes[2];
+        } else {
+            major_version = unexpected_version_num;
+            middle_version = unexpected_version_num;
+            mini_version = unexpected_version_num;
+        }
+    }
+
+    bool
+    Valid() {
+        return major_version != unexpected_version_num && middle_version != unexpected_version_num &&
+               mini_version != unexpected_version_num;
+    };
+
+    std::string
+    VersionCode() {
+        return version_code;
+    }
+
+    static bool
+    VersionCheck(std::string version) {
+        try {
+            return std::regex_match(version.c_str(), std::regex(version_regex));
+        } catch (std::regex_error& e) {
+            LOG_KNOWHERE_ERROR_ << "unexpected index version : " << version;
+        }
+        return false;
+    }
+
+    // used when version is not set
+    static inline Version
+    GetDefaultVersion() {
+        return Version(default_version);
+    }
+
+    // the current version (newest version support)
+    static inline Version
+    GetCurrentVersion() {
+        return Version(current_version);
+    }
+
+    // the minimal version (oldest version support)
+    static inline Version
```

Comment:
why need a min version ?

Reply:
Minimal version gives the oldest version which knowhere could support.